### DISCUSSION TITLE
Train Segmentation and Embedding Model From Scratch

### DISCUSSION THREAD

**Discussion by eggplant95** [juanmc2005/diart#125] [Q&A]
Hi, Thank you for your awsome contribution. 

I have learnt how to finetune your pretrained segmentation model in https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/training_a_model.ipynb. However, it didn't mention  how to train a segmentation from scratch. 
I would like to know how could I reproduce your work with the listed data in your paper? Futhermore, how could I train a model from scratch with my own data? Is there any pipeline about the training details?

I would be much appreciated if you could tell me more about these details. Thank you very much.

**Comment by juanmc2005** [comment]
Hi @eggplant95,

> I would like to know how could I reproduce your work with the listed data in your paper?

You need to download both audio and RTTM files for the datasets we used (we are not allowed to provide them ourselves) and then run `diart.benchmark`.
Notice that you probably won't get the exact same results because I've been working on some small improvements since the paper got published.

> Futhermore, how could I train a model from scratch with my own data? Is there any pipeline about the training details?

To train a segmentation model from scratch you can follow the part of the tutorial with the title "Training a voice activity detection model from scratch", but replacing the `VoiceActivityDetection` task with `Segmentation`. The architecture used in the tutorial is the same as `pyannote/segmentation`.
The training configuration is available [here](https://huggingface.co/pyannote/segmentation/blob/main/config.yaml), other hyper-parameters like learning rate and batch size will depend on your own data and hardware.

**Reply by m15kh** [reply]
hi @juanmc2005  
 i trained model in pyannote and i have a checkpoint so you didn't explain how can i useing my checkpoint in your code!
 can you say example for that
 thanks for your answering
